src/control/directive.ts

```diff
@@ -188,31 +191,35 @@ export class NgrxFormControlDirective<TStateValue extends FormControlValueTypes,
     this.updateViewIfIsDisabledChanged(this.state, undefined);
     this.updateViewIfIsFocusedChanged(this.state, undefined);
 
+    const dispatchMarkAsDirtyAction = () => {
+      if (this.state.isPristine) {
+        this.actionsSubject.next(new MarkAsDirtyAction(this.state.id));
+      }
+    };
+
     const dispatchSetValueAction = () => {
       this.stateValue = this.ngrxValueConverter.convertViewToStateValue(this.viewValue);
       if (this.stateValue !== this.state.value) {
         this.actionsSubject.next(new SetValueAction(this.state.id, this.stateValue));
 
-        if (this.state.isPristine) {
-          this.actionsSubject.next(new MarkAsDirtyAction(this.state.id));
-        }
+        dispatchMarkAsDirtyAction();
       }
     };
 
     this.viewAdapter.setOnChangeCallback((viewValue: TViewValue) => {
       this.viewValue = viewValue;
 
-      if (this.ngrxUpdateOn === CHANGE) {
+      if (this.ngrxUpdateOn === NGRX_UPDATE_ON_TYPE.CHANGE) {
         dispatchSetValueAction();
       }
     });
 
     this.viewAdapter.setOnTouchedCallback(() => {
-      if (!this.state.isTouched) {
+      if (!this.state.isTouched && this.ngrxUpdateOn !== NGRX_UPDATE_ON_TYPE.NEVER) {
```

Comment:
This line is never hit by your test since you are marking the state already as touched. See my comment in the test file.

Reply:
Yeah Ive change it in tests file.